### ISSUE TITLE
Write capabilities

### ISSUE THREAD

**Issue by spotlightishere** [wii-tools/arclib#1]
As of writing, creating a U8 archive isn't possible. This should be implemented.

**Comment by ghost** [comment]
Well Spotlight, I could temporaily set this to use Larsen's python implementation via a subprocess syscall.
```
import sentry_sdk as sentry
import struct as packs
import logging as logs
import copy as cp
from sentry.integrations.logging import LoggingIntegration
#By 6100m and Larsen
#And don't call us nezzie ( ͡° ͜ʖ ͡°)
t=True #T for True, just like vintage programs used to do.
f=False #F for False, long live DOS prompts!
num=0
misc=4294967295
func=copy.copy(num)
production=f
perrors=f
def setuplog(sentryurl,printerrors):
    #We could do Sentry if we wanted. I mean, I installed Sentry on our Cacti VPS, so Ig we can use that?
    global logger,production
    sentry_logs=LoggingIntegration(
        level=logs.INFO,
        event_level=logs.INFO
    )
    sentry.init(dsn=sentryurl,integrations=[sentry_logging])
    logger=logs.getLogger(__name__)
    perrors=printerrors
    production=t
def log(msg,level):
    if perrors:
        print(msg)
    if production:
        if level==1: #To Larsen: Yes, as soon as I can, I will get number levels PR'd to RC24's File-Maker repo.
            logger.debug(msg) #Debug Message Mode
        elif level==2:
            logger.info(msg) #Info Message Mode
        elif level==3:
            logger.warning(msg) #Warning Message Mode
        elif level==4:
            logger.critical(msg) #Critical Message Mode
def u8(data):
    if not 0<=data<=255:
        log("u8 out of range: %s" % data,2)
        data=func()
    return packs.pack(">B",data)
def u16(data):
    if not 0<=data<=65535:
        log("u16 out of range: %s" % data,2)
        data=func()
    return packs.pack(">H",data)
def u32(data):
    if not 0<=data<=misc:
        log("u32 out of range: %s" % data,2)
        data=func()
    return packs.pack(">I",data)
def u32_littleendian(data):
    if not 0<=data<=misc:
        log("u32 little endian out of range: %s" % data,2)
        data=func()
    return packs.pack("<I",data)
``` 
but I could defintely see if there is a way to get this put into Go, I'm certain there is. (That, or I could just grab the implementation out of Mail-Go, as I think it has a Go implementation of this.)

**Comment by spotlightishere** [comment]
Partial support was committed in 5664b0f91af5164d2d3d12b7c76d7870852bc856, but all files extracted appear to be one-off.

**Comment by spotlightishere** [comment]
The aforementioned issue was resolved in 4f2e159d33bb29a234b42c7c53eaa341055b9ffe. Parent directories were being moved from being tracked before their files could be attached, causing all files to eventually find themselves in two directories above.
